crates/sp-domains-fraud-proof/src/tests.rs

```diff
@@ -39,6 +41,556 @@ fn generate_storage_proof_for_tx_validity(
+        );
+        alice_nonce += 1;
+
+        let transfer_with_big_tip_2 = construct_extrinsic_generic::<TestRuntime, _>(
+            &alice.client,
+            pallet_balances::Call::transfer_allow_death {
+                dest: Charlie.to_account_id(),
+                value: 1,
+            },
+            alice.key,
+            false,
+            alice_nonce,
+            4000000000u32 / 3,
+        );
+
+        if commit_mode {
+            alice_nonce += 1;
+        }
+
+        let transfer_with_big_tip_3 = construct_extrinsic_generic::<TestRuntime, _>(
+            &alice.client,
+            pallet_balances::Call::transfer_allow_death {
+                dest: Charlie.to_account_id(),
+                value: 1,
+            },
+            alice.key,
+            false,
+            alice_nonce,
+            4000000000u32 / 3,
+        );
+
+        assert!(runtime_instance
+            .check_transaction_and_do_pre_dispatch(
+                alice.client.as_ref().info().best_hash,
+                &transfer_with_big_tip_1.clone().into(),
+                alice.client.as_ref().info().best_number,
+                alice.client.as_ref().info().best_hash
+            )
+            .unwrap()
+            .is_ok());
+
+        assert!(runtime_instance
+            .execute_in_transaction(|api| {
+                if commit_mode {
+                    TransactionOutcome::Commit(api.check_transaction_and_do_pre_dispatch(
+                        alice.client.as_ref().info().best_hash,
+                        &transfer_with_big_tip_2.clone().into(),
+                        alice.client.as_ref().info().best_number,
+                        alice.client.as_ref().info().best_hash,
+                    ))
+                } else {
+                    TransactionOutcome::Rollback(api.check_transaction_and_do_pre_dispatch(
+                        alice.client.as_ref().info().best_hash,
+                        &transfer_with_big_tip_2.clone().into(),
+                        alice.client.as_ref().info().best_number,
+                        alice.client.as_ref().info().best_hash,
+                    ))
+                }
+            })
+            .is_ok());
+
+        assert_eq!(
+            runtime_instance
+                .check_transaction_and_do_pre_dispatch(
+                    alice.client.as_ref().info().best_hash,
+                    &transfer_with_big_tip_3.clone().into(),
+                    alice.client.as_ref().info().best_number,
+                    alice.client.as_ref().info().best_hash
+                )
+                .unwrap(),
+            if commit_mode {
+                Err(
+                    CheckTxValidityError::decode(&mut tx_validity_error.encode().as_slice())
+                        .unwrap(),
+                )
+            } else {
+                Ok(())
+            }
+        )
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn check_bundle_validity_runtime_api_should_work() {
```

Comment:
It seems this check is duplicated with `runtime_instance_assumptions_are_correct`

Reply:
The tests are mostly similar, but the runtime assumption test is there to make sure that the assumptions we made throughout the code regarding runtime api is valid. 

While this test is more concerned about testing the api.